packages/experimental/Avatar/src/JSAvatar.tsx

```diff
@@ -32,29 +36,32 @@ export const JSAvatar = compose<JSAvatarType>({
     image: Image,
     initials: Text,
     initialsBackground: View,
-    icon: Image,
+    icon: Icon,
     ring: View,
     badge: PresenceBadge,
   },
   useRender: (userProps: JSAvatarProps, useSlots: UseSlots<JSAvatarType>) => {
     const avatar = useAvatar(userProps);
     const Slots = useSlots(userProps, (layer) => avatarLookup(layer, avatar.state, userProps));
+    const iconProps = createIconProps(userProps.icon);
 
     return (final: JSAvatarProps) => {
-      const { activeAppearance, initials, image, badge, ...mergedProps } = mergeProps(avatar.props, final);
+      const { activeAppearance, icon, initials, image, badge, ...mergedProps } = mergeProps(avatar.props, final);
       const { showRing, transparentRing } = avatar.state;
 
       return (
         <Slots.root {...mergedProps}>
           {image.source ? (
             <Slots.image {...image} />
-          ) : (
+          ) : initials ? (
             <Slots.initialsBackground>
               <Slots.initials>{initials}</Slots.initials>
             </Slots.initialsBackground>
+          ) : (
+            <Slots.icon {...iconProps} />
           )}
           {showRing && !transparentRing && <Slots.ring />}
-          <Slots.badge {...badge} />
+          {badge.status ? <Slots.badge {...badge} /> : null}
```

Comment:
nit: I think this could be written as `{badge.status && <Slots.badge {...badge} />}`